protocols/gossipsub/src/protocol.rs

```diff
@@ -375,7 +375,9 @@ impl Decoder for GossipsubCodec {
             // ensure the sequence number is a u64
             let sequence_number = if verify_sequence_no {
                 if let Some(seq_no) = message.seqno {
-                    if seq_no.len() != 8 {
+                    if seq_no.len() == 0 {
```

Comment:
Hmmm... note that this check is only active for ValidationMode `Strict` and `Permissive`, so it is not really relevant to ETH2, I changed that before we switched to Anonymous because in protobuf a missing value is equivalent to its default (which is in case of bytes the zero byte value).